packages/compass-preferences-model/src/renderer-ipc.ts

```diff
@@ -0,0 +1,52 @@
+import ipc from 'hadron-ipc';
+import type {
+  GlobalPreferences,
+  UserPreferences,
+  OnPreferenceValueChangedCallback,
+} from './preferences';
+
+/**
+ * API to communicate with preferences from the electron renderer process.
+ */
+export const preferencesIpc = {
+  savePreferences(attributes: Partial<GlobalPreferences>) {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:save-preferences', attributes);
+    }
+    return {} as UserPreferences;
+  },
+  getPreferences() {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:get-preferences');
+    }
+    return {} as GlobalPreferences;
+  },
+  getConfigurableUserPreferences() {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke(
+        'compass:get-configurable-user-preferences'
+      );
+    }
+    return {} as UserPreferences;
+  },
+  onPreferenceValueChanged(
+    preferenceName: string,
+    callback: OnPreferenceValueChangedCallback
+  ) {
```

Comment:
```suggestion
  onPreferenceValueChanged<K extends keyof GlobalPreferences>(
    preferenceName: K,
    callback: OnPreferenceValueChangedCallback<GlobalPreferences[K]>
  ): () => void {
```

or, if you feel like that’s less involved, without `OnPreferenceValueChangedCallback` at all:

```suggestion
  onPreferenceValueChanged<K extends keyof GlobalPreferences>(
    preferenceName: K,
    callback: (value: GlobalPreferences[K]) => void
  ): () => void {
```